WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/Purchase Note/ProductPurchaseNoteViewController.swift

```diff
@@ -140,6 +155,17 @@ private extension ProductPurchaseNoteViewController {
             self?.productSettings.purchaseNote = text
         }
     }
+
+    func getIndexPathForRow(_ row: Row) -> IndexPath? {
```

Comment:
this seems to be a frequently used utility that could be shared, this is one way I could think of:

```
protocol RowIterable {
    associatedtype Row: Equatable
    var rows: [Row] { get }
}

extension Array where Element: RowIterable {
    func getIndexPathForRow(_ row: Element.Row) -> IndexPath? {
        for s in 0 ..< count {
            let rows = self[s].rows
            for r in 0 ..< rows.count {
                if rows[r] == row {
                    return IndexPath(row: r, section: s)
                }
            }
        }
        return nil
    }
}
```

and each `Section` can conform to `RowIterable`. what do you think?

Reply:
I like this improvement, since we start to use this method in many points of the app, and we have duplicated code.
Since this touches not only these classes but also the Product Inventory, I opened an issue here https://github.com/woocommerce/woocommerce-ios/issues/2161 and I'll open a separate PR for this implementation 👍